src/EditorFeatures/Core/Implementation/IntelliSense/SignatureHelp/Controller.Session_ComputeModel.cs

```diff
@@ -136,25 +103,25 @@ private static bool SequenceEquals(IEnumerable<string> s1, IEnumerable<string> s
                 return s1 != null && s2 != null && s1.SequenceEqual(s2);
             }
 
-            private static SignatureHelpItem GetSelectedItem(Model currentModel, SignatureHelpItems items, ISignatureHelpProvider provider)
+            private static SignatureHelpItem GetSelectedItem(Model currentModel, SignatureList signatureList)
             {
                 // Try to find the most appropriate item in the list to select by default.
 
                 // If the provider specified one a selected item, then always stick with that one. 
-                if (items.SelectedItemIndex.HasValue)
+                if (signatureList.SelectedItemIndex.HasValue)
                 {
-                    return items.Items[items.SelectedItemIndex.Value];
+                    return signatureList.Items[signatureList.SelectedItemIndex.Value];
                 }
 
                 // If the provider did not pick a default, and it's the same provider as the previous
                 // model we have, then try to return the same item that we had before. 
-                if (currentModel != null && currentModel.Provider == provider)
```

Comment:
I removed it because I'm trying to remove the need for the `Provider` completely, but the work is still incomplete. Removing this did result in any failing tests.